app/services/apply_api/import_application.rb

```diff
@@ -11,9 +11,7 @@ def initialize(application_data:)
     end
 
     def call
-      return unless provider
-
-      application.update!(application: application_data.to_json, provider: provider, state: state)
+      application.update!(application: application_data.to_json, provider_code: provider_code, state: state)
```

Comment:
Should we use the state methods rather than setting it directly? Just thinking if we add any logic to the transition at some point it might get accidentally skipped here.

What about something like - 

```
ApplyApplication.transaction do
  application.update!(...
  hei_provider? ? application.non_importable_hei! : application.importable!
end
```

Reply:
I was thinking of doing this but then I thought maybe it was better to have one DB call instead of two. I'm happy you brought it up because we are leaking state data into other objects. Two DB calls is not th end of the world.